PagingWithNetworkSample/app/src/main/java/com/android/example/paging/pagingwithnetwork/reddit/ui/RedditActivity.kt

```diff
@@ -78,34 +80,38 @@ class RedditActivity : AppCompatActivity() {
         initSearch()
     }
 
+    override fun onDestroy() {
+        super.onDestroy()
+        refreshListener?.let { (list.adapter as? PostsAdapter)?.removeLoadStateListener(it) }
```

Comment:
why do you need this cleanup?

The Activity has the same lifecycle as the PostsAdapter.

Reply:
Done